src/main/java/com/example/damian/iracki/ticketBookingApp/resources/TicketResource.java

```diff
@@ -0,0 +1,94 @@
+package com.example.damian.iracki.ticketBookingApp.resources;
+
+
+import com.example.damian.iracki.ticketBookingApp.dto.TicketDto;
+import com.example.damian.iracki.ticketBookingApp.entities.Screening;
+import com.example.damian.iracki.ticketBookingApp.entities.Ticket;
+import com.example.damian.iracki.ticketBookingApp.services.ScreeningService;
+import com.example.damian.iracki.ticketBookingApp.services.TicketService;
+import com.example.damian.iracki.ticketBookingApp.validators.TicketValidator;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@RestController
+@RequestMapping("/tickets")
+public class TicketResource {
+
+    private final TicketService ticketService;
+    private final ScreeningService screeningService;
+
+    @Autowired
+    public TicketResource(TicketService ticketService, ScreeningService screeningService) {
+        this.ticketService = ticketService;
+        this.screeningService = screeningService;
+    }
+
+    @GetMapping("/")
+    public ResponseEntity<List<Ticket>> getAllTickets() {
+        List<Ticket> tickets = ticketService.getAllTickets();
+        return new ResponseEntity<>(tickets, HttpStatus.OK);
+    }
+
+    @Transactional
+    @PostMapping("/")
+    public String addTickets(@RequestBody List<TicketDto> ticketDtos){
+        if(ticketDtos.size() < 1){
+            throw  new IllegalStateException("You must book one ticket at least");
+        }
+        List<Ticket> tickets = mapAllTicketDtoToTicket(ticketDtos);
+        Screening screening = screeningService.findScreeningById(tickets.get(0).getScreeningId());
+        TicketValidator ticketValidator = new TicketValidator();
+        for (Ticket ticket : tickets) {
+            if(ticketValidator.checkTicket(ticket, screening, LocalDateTime.now())){
```

Comment:
Czy istnieje możliwość, żeby warunek w tym `if`ie kiedykolwiek zewaluował się na `false`?

Reply:
Tak, jeżeli z frontu dostaniemy niepoprawne dane.

Reply:
W zasadzie to nie, ponieważ pojedyncze metody w walidatorze zwracają true  lub wyjątek, nigdy false.